### PR TITLE
feat(FN-1725): add sql id path parameter test helper

### PR DESCRIPTION
**Pull request by JoshBinns2000**
## Introduction :pencil2:
We want to add a test helper for testing the sql id path parameter checking middleware.

## Resolution :heavy_check_mark:
- Updates the sql id middleware to respond with an `ApiErrorResponse`
- Adds a new error code for invalid sql id path parameters
- Updates dtfs-central-api api-tests to use the new test helper

### CODE REVIEW

File: dtfs-central-api/api-tests/v1/utilisation-reports/put-utilisation-report-status.api-test.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { SqlDbHelper } from '../../sql-db-helper';
 
 console.error = jest.fn();
 
-describe('/v1/utilisation-reports/set-status', () => {
+describe('PUT /v1/utilisation-reports/set-status', () => {
   const setStatusUrl = '/v1/utilisation-reports/set-status';
```
Comment (sw-mattw) [inline dtfs-central-api/api-tests/v1/utilisation-reports/put-utilisation-report-status.api-test.ts:9]: Nitpick, could we pull this out into a BASE_URL var and use it in the description text too (as with the other test files)

**Review by sw-mattw — APPROVED** [review]
Nice, very tidy. Like the use of BASE_URL and replacing the parameters in this.

One minor nitpick, up to you if it's changed, approved anyway

File: dtfs-central-api/api-tests/v1/utilisation-reports/put-utilisation-report-status.api-test.ts (hunk 1 above)
Comment (JoshBinns2000) [reply dtfs-central-api/api-tests/v1/utilisation-reports/put-utilisation-report-status.api-test.ts:9]: Done :)

Comment (sw-mattw) [reply dtfs-central-api/api-tests/v1/utilisation-reports/put-utilisation-report-status.api-test.ts:11]: Sorry as in once we've pulled BASE_URL out could we use that both in the description text (done now) and also replace all `setStatusUrl` references with BASE_URL and remove the `setStatusUrl` var

**Review by sw-mattw — APPROVED** [review]
Looks good, thanks for the change

**Pull request merged**